benchmarks/linalg_refactor_benchmark.cpp

```diff
@@ -0,0 +1,151 @@
+#include <shogun/base/init.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/linalgrefactor/linalgRefactor.h>
+
+#include <shogun/mathematics/eigen3.h>
+#include <viennacl/linalg/inner_prod.hpp>
+#include <viennacl/vector.hpp>
+
+#include <algorithm>
+#include <memory>
+#include <hayai/hayai.hpp>
+
+#include <iostream>
+
+using namespace shogun;
+
+/**
+ * Instructions :
+ * 1. Install benchmarking toolkit "hayai" (https://github.com/nickbruun/hayai)
+ * 2. Compile against libhayai_main, e.g.
+ * 		g++ -O3 -std=c++11 linglg_refactor_benchmark.cpp -I/usr/include/eigen3 \
+ *		-lshogun -lhayai_main -lOpenCL -o benchmark
+ * 3. ./benchmark
+ */
+
+/** Generate data only once */
+typedef viennacl::vector_base<int32_t, std::size_t, std::ptrdiff_t> VCLVectorBase;
+
+struct Data
+{
+	typedef viennacl::backend::mem_handle VCLMemoryArray;
+
+	Data()
+	{
+		num_rows = 100;
+		A = init(1);
+		B = init(2);
+		Ac = init_c(A);
+		Bc = init_c(B);
+        Ag = init_g(A);
+        Bg = init_g(B);
+		Av = init_v(A);
+		Bv = init_v(B);
+		init_shogun_with_defaults();
+	}
+
+	Data(index_t num_rows)
+	{
+		this->num_rows = num_rows;
+		A = init(1);
+		B = init(2);
+		Ac = init_c(A);
+		Bc = init_c(B);
+		Ag = init_g(A);
+		Bg = init_g(B);
+		Av = init_v(A);
+		Bv = init_v(B);
+		init_shogun_with_defaults();
+	}
```

Comment:
You could have just created a `init()` method that does all those stuff, and then could have called that method from both the constructor. Anyway, not a big deal